### PR TITLE
db: make schema commitlog feature mandatory

### PR DESCRIPTION
**Pull request by patjed41**
Using consistent cluster management and not using schema commitlog
ends with a bad configuration throw during bootstrap. Soon, we
will make consistent cluster management mandatory. This forces us
to also make schema commitlog mandatory, which we do in this patch.

A booting node decides to use schema commitlog if at least one of
the two statements below is true:
- the node has `force_schema_commitlog=true` config,
- the node knows that the cluster supports the `SCHEMA_COMMITLOG`
  cluster feature.

The `SCHEMA_COMMITLOG` cluster feature has been added in version
5.1. This patch is supposed to be a part of version 6.0. We don't
support a direct upgrade from 5.1 to 6.0 because it skips two
versions - 5.2 and 5.4. So, in a supported upgrade we can assume
that the version which we upgrade from has schema commitlog. This
means that we don't need to check the `SCHEMA_COMMITLOG` feature
during an upgrade.

The reasoning above also applies to Scylla Enterprise. Version
2024.2 will be based on 6.0. Probably, we will only support
an upgrade to 2024.2 from 2024.1, which is based on 5.4. But even
if we support an upgrade from 2023.x, this patch won't break
anything because 2023.1 is based on 5.2, which has schema
commitlog. Upgrades from 2022.x definitely won't be supported.

When we populate a new cluster, we can use the
`force_schema_commitlog=true` config to use schema commitlog
unconditionally. Then, the cluster feature check is irrelevant.
This check could fail because we initiate schema commitlog before
we learn about the features. The `force_schema_commitlog=true`
config is especially useful when we want to use consistent cluster
management. Failing feature checks would lead to crashes during
initial bootstraps. Moreover, there is no point in creating a new
cluster with `consistent_cluster_management=true` and
`force_schema_commitlog=false`. It would just cause some initial
bootstraps to fail, and after successful restarts, the result would
be the same as if we used `force_schema_commitlog=true` from the
start.

In conclusion, we can unconditionally use schema commitlog without
any checks in 6.0 because we can always safely upgrade a cluster
and start a new cluster.

Apart from making schema commitlog mandatory, this patch adds two
changes that are its consequences:
- making the unneeded `force_schema_commitlog` config unused,
- deprecating the `SCHEMA_COMMITLOG` feature, which is always
  assumed to be true.

### CODE REVIEW

**Comment by kbr-scylla** [comment]
@tgrabiec @gusev-p please review the last commit (` db: always use schema commitlog `)

File: db/config.cc [hunk 1]
```diff
@@ -1083,8 +1083,6 @@ db::config::config(std::shared_ptr<db::extensions> exts)
     , restrict_future_timestamp(this, "restrict_future_timestamp",liveness::LiveUpdate, value_status::Used, true, "Controls whether to detect and forbid unreasonable USING TIMESTAMP, more than 3 days into the future.")
     , ignore_truncation_record(this, "unsafe_ignore_truncation_record", value_status::Used, false,
         "Ignore truncation record stored in system tables as if tables were never truncated.")
-    , force_schema_commit_log(this, "force_schema_commit_log", value_status::Used, false,
```
Comment (kbr-scylla) [inline db/config.cc:1086]: I think the usual way is to modify an option to `value_status::Unused` instead of deleting it completely. Otherwise, won't the node complain on startup (maybe even fail startup) if conf/scylla.yaml contains unknown option (I'm not sure) -- which may happen when upgrading from older versions?

File: conf/scylla.yaml [hunk 2]
```diff
@@ -559,18 +559,6 @@ murmur3_partitioner_ignore_msb_bits: 12
 # Set to `false` to fall-back to the old algorithm.
```
[the remark this answers is not in the document]
Comment (kbr-scylla) [reply conf/scylla.yaml:559]: This table may be helpful: https://github.com/scylladb/scylla-enterprise/wiki/Currently-maintained-Open-source-and-Enterprise-branches#enterprise

And the Scylla Enterprise upgrade docs: enterprise->enterprise: https://enterprise.docs.scylladb.com/branch-2023.1/upgrade/upgrade-enterprise/index.html

and OSS->enterprise: https://enterprise.docs.scylladb.com/branch-2023.1/upgrade/upgrade-to-enterprise/index.html

File: db/config.cc (hunk 1 above)
Comment (patjed41) [reply db/config.cc:1090]: Fixed.

File: conf/scylla.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (patjed41) [reply conf/scylla.yaml:559]: I included a consideration.

**Comment by patjed41** [comment]
v2:
- included a consideration about version upgrades in Scylla Enterprise in PR header,
- made `force_schema_commit_log` unused instead of completely removed,
- removed `force_schema_commit_log` from a config created in `scylla_cluster.py` (I missed it previously)

**Comment by patjed41** [comment]
v3: clarified title

**Review by avikivity — APPROVED** [review]
